### PR TITLE
Fix open source logging. Move logging implementation to a cc file.

### PR DESCRIPTION
**Pull request by kuhar**
Fix open source logging. Move logging implementation to a cc file.

Introduce a simple helper class for logging. This makes sure that
the concrete logging library is hidden from layer files that can't
depend on implementation details.

Include source file names and line numbers in open source logs.

This depends on https://github.com/googlestadia/performance-layers/pull/30 and needs to be rebased before submitting.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Copyright 2020 Google LLC
+# Copyright 2020-2021 Google LLC
```
Comment (dnovillo) [inline CMakeLists.txt]: This is from #30 right?

File: layer/layer_utils.h [hunk 2]
```diff
@@ -15,11 +15,47 @@
...
+namespace performancelayers {
+enum class LogMessageKind { INFO, WARNING, ERROR };
+
+// Helper class for printing log messages. Exposes a common logging macro,
+// SPL_LOG, independent of the underlying logging library. Sample use:
+//   SPL_LOG(WARNING) << "Cannot load file: " << my_file_path;
+//
+// Note that there is no need to add a newline character at the end -- this is
+// handled by the implementation.
+class MessageLogger {
```
Comment (dnovillo) [inline layer/layer_utils.h]: nit: should this be in a `layers/log.h` file instead?  To prevent accumulating unrelated stuff in this generic file.

File: layer/runtime_layer.cc [hunk 3]
```diff
@@ -12,14 +12,14 @@
...
 
-#include "runtime_layer_data.h"
-
 #include <cassert>
 #include <cstdint>
 #include <cstring>
 #include <functional>
 #include <string>
 
+#include "runtime_layer_data.h"
```
Comment (dnovillo) [inline layer/runtime_layer.cc:21]: What does this include bring?  It does not seem to be used.

**Review by dnovillo — CHANGES_REQUESTED** [review]

File: layer/runtime_layer.cc (hunk 3 above)
Comment (kuhar) [reply layer/runtime_layer.cc:21]: Clang-format reordered includes. `RuntimeLayerData` is used in almost every function.

File: layer/layer_utils.h [hunk 4]
```diff
@@ -15,11 +15,4 @@
 #ifndef STADIA_OPEN_SOURCE_PERFORMANCE_LAYERS_UTILS_H_
 #define STADIA_OPEN_SOURCE_PERFORMANCE_LAYERS_UTILS_H_
```
Comment (kuhar) [inline layer/layer_utils.h:16]: This header is empty now, but I'll layer macros here in a next PR. I didn't remove it entirely because of that.

File: layer/layer_utils.h (hunk 2 above)
Comment (kuhar) [reply layer/layer_utils.h]: Changed this to logging.h. Thanks for the suggestion.

**Review by dnovillo — APPROVED** [review]
Thanks!  LGTM.
